brave/brave-core
# (Wallet) Implement new password creation screen

## Resolves https://github.com/brave/brave-browser/issues/36552

Implements new password creation screen following [Figma design](https://www.figma.com/file/oXIiv45pIhF2RGlpEjOfUg/Wallet-onboarding?node-id=6223%3A70942&mode=dev).

Main umbrella issue: https://github.com/brave/brave-browser/issues/35294


https://github.com/brave/brave-core/assets/3308503/e6d7fd79-9aec-48eb-bb40-12e43a57ebda

### Notable Changes

- Implements responsive layout that adjust continue button position with and without software keyboard.
- Implements stronger checks that prevents multiple tappings on the buttons loading the same fragment twice.
- Moves wallet creation logic into appropriate fragment `OnboardingCreatingWalletFragment`.
- Implements new UI component `PasswordStrengthMeterView` following latest Material guidelines.
- Migrate away from deprecated `FragmentStatePagerAdapter` in favor of newer `FragmentStateAdapter` and `ViewPager2` bringing better perfomance and removing callback duplication.


### Light Mode

 <img src="https://github.com/brave/brave-core/assets/3308503/48f7ba16-9ffb-409c-9799-83761a244b4c" width="500">

### Dark Mode

 <img src="https://github.com/brave/brave-core/assets/3308503/a9f1d0c8-4675-4b68-a019-440c061f5d1d" width="500">



## Submitter Checklist:

  * `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests` [wiki](https://github.com/brave/brave-browser/wiki/Tests)
  * `npm run presubmit` [wiki](https://github.com/brave/brave-browser/wiki/Presubmit-checks), `npm run gn_check`, `npm run tslint`

## Reviewer Checklist:


## After-merge Checklist:

  changes has landed on

## Test Plan:

- Create a new Wallet
- Accept terms of use
- Enter a password in the first edit text
- Re-type the password in the second edit text
- Observe the match sign pops up and the button becomes enabled
- Modify the password in the first edit text
- Observe the button becames disabled
- Increase password length
- Observe weak, medium, and strong match the logic of Desktop
- Try to tap on continue button multiple times
- Observe the button can be tapped only when the passwords match and only a single time
